### DISCUSSION TITLE
Can I use useNavigate to convert search params into ampersand? e.g. filter=apple&filter=orange instead of filter=apple%2Corange?

### DISCUSSION THREAD

**Discussion by starxyn** [TanStack/router#2269] [Q&A] [answered]
```
import { useNavigate } from '@tanstack/react-router';

const basket = {
  type: "fruits",
  filter: ["sour, sweet"],
};
 navigate({
   to: '/abcmart',
   params: {
     abc       
   },
   search: {
     type: type,
     filter: filter,
    },
   });
}
```
note: tanstack router version is 1.26.1
but this doesn't work. it just auto converts to http://abcmart?type=type&filter=sour%2Csweet. All I wanted is http://abcmart?type=type&filter=sour&filter=sweet. i tried other string manipulation method it also auto converts to other kind of weird query string. what can I do?

**Comment by schiller-manuel** [comment]
yes this can be done using *Custom Search Param Serialization*. checkout the section about the `query-string` library

https://tanstack.com/router/v1/docs/framework/react/guide/custom-search-param-serialization#using-the-query-string-library

**Reply by starxyn** [reply]
But it still doesn't turn the array values into ampersand. What I mean is to achieve http://abcmart?type=type&filter=sour&filter=sweet, not http://abcmart?type=type&filter=sour%2Csweet](http://abcmart/?type=type&filter=sour%2Csweet. Note the filter: ['sour','sweet'] is split into filter=sour&filter=sweet.

**Reply by schiller-manuel** [reply]
this setup works:

```
const router = createRouter({
  routeTree,
  defaultPreload: 'intent',
  stringifySearch: (value) => {
    const searchStr = qs.stringify(value);
    return searchStr ? `?${searchStr}` : '';
  },
  parseSearch: qs.parse,
});
```

see full example: https://stackblitz.com/edit/tanstack-router-wewitc?file=src%2Fmain.tsx,src%2Froutes%2Findex.tsx&preset=node

**Reply by starxyn** [reply]
So this means that using useNavigate will not work, but using Link will work?

**Reply by starxyn** [reply]
But I have to use useNavigate because I'm not using an button click event to direct to a website. useNavigate is part of a backend logic to automatically direct to a website.

**Reply by schiller-manuel** [reply, marked as answer]
useNavigate and Link share the same API, so if you change the search param validation it affects both

**Reply by starxyn** [reply]
thank you!
